src/libqof/qof/gnc-numeric.hpp

```diff
@@ -0,0 +1,418 @@
+            throw std::invalid_argument("Attempt to construct a GncNumeric with a 0 denominator.");
+        /* gnc_numeric has a dumb convention that a negative denominator means
+         * to multiply the numerator by the denominator instead of dividing.
+         */
+        if (in.denom < 0)
+        {
+            m_num *= -in.denom;
+            m_den = 1;
+        }
+    }
+    /**
+     * Double constructor.
+     *
+     * @param d The double to be converted. In order to fit in an int64_t, its
+     * absolute value must be < 1e18; if its absolute value is < 1e-18 it will
+     * be represented as 0, though for practical purposes nearly all commodities
+     * will round to zero at 1e-9 or larger.
+     */
+    GncNumeric(double d);
+
+    /**
+     * String constructor.
+     *
+     * Accepts integer values in decimal and hexadecimal. Does not accept
+     * thousands separators. If the string contains a '/' it is taken to
+     * separate the numerator and denominator; if it conains either a '.' or a
+     * ',' it is taken as a decimal point and the integers on either side will
+     * be combined and a denominator will be the appropriate power of 10. If
+     * neither is present the number will be treated as an integer and m_den
+     * will be set to 1.
+     *
+     * Whitespace around a '/' is ignored. A correctly-formatted number will be
+     * extracted from a larger string.
+     *
+     * Numbers that cannot be represented with int64_ts will throw
+     * std::out_of_range unless a decimal point is found (see above). A 0
+     * denominator will cause the constructor to throw std::underflow_error. An
+     * empty string or one which contains no recognizable number will result in
+     * std::invalid_argument.
+     */
+    GncNumeric(const std::string& str, bool autoround=false);
+    GncNumeric(const GncNumeric& rhs) = default;
+    GncNumeric(GncNumeric&& rhs) = default;
+    GncNumeric& operator=(const GncNumeric& rhs) = default;
+    GncNumeric& operator=(GncNumeric&& rhs) = default;
+    ~GncNumeric() = default;
+    /**
+     * gnc_numeric conversion. Use static_cast<gnc_numeric>(foo)
+     */
+    operator gnc_numeric() const noexcept;
+    /**
+     * gnc_numeric conversion. Use static_cast<double>(foo)
+     */
+    operator double() const noexcept;
+
+    /**
+     * Accessor for numerator value.
+     */
+    int64_t num() const noexcept { return m_num; }
+    /**
+     * Accessor for denominator value.
+     */
+    int64_t denom() const noexcept { return m_den; }
+    /**
+     * @return A GncNumeric with the opposite sign.
+     */
+    GncNumeric operator-() const noexcept;
+    /**
+     * @return 0 if this == 0 else 1/this.
+     */
+    GncNumeric inv() const noexcept;
+    /**
+     * @return -this if this < 0 else this.
+     */
+    GncNumeric abs() const noexcept;
+    /**
+     * Reduce this to an equivalent fraction with the least common multiple as
+     * the denominator.
```

Comment:
How about "Return an equivalent fraction with all common factors between the numerator and denominator removed." ?